SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -410,18 +410,20 @@ void USpatialNetDriver::CreateAndInitializeCoreClasses()
 	{
 		VirtualWorkerTranslator = MakeUnique<SpatialVirtualWorkerTranslator>();
 		VirtualWorkerTranslator->Init(this);
-		LoadBalanceEnforcer = NewObject<USpatialLoadBalanceEnforcer>();
-		LoadBalanceEnforcer->Init(Connection->GetWorkerId(), StaticComponentView, Sender, VirtualWorkerTranslator.Get());
 
 		// TODO: zoning - Move to AWorldSettings subclass [UNR-2386]
+		// TODO: Exit cleanly if SpatialSettings->LoadBalanceStrategy is not set.
 		LoadBalanceStrategy = NewObject<UAbstractLBStrategy>(this, SpatialSettings->LoadBalanceStrategy);
 		LoadBalanceStrategy->Init(this);
 
-		VirtualWorkerTranslator->SetDesiredVirtualWorkerCount(LoadBalanceStrategy->GetVirtualWorkerIds().Num());
+		VirtualWorkerTranslator->AddVirtualWorkerIds(LoadBalanceStrategy->GetVirtualWorkerIds());
```

Comment:
We only need to do this on the authoritative worker? Can we move this to inside `SpatialVirtualWorkerTranslator::AuthorityChanged`? (it has a pointer to NetDriver so should be fine?)

Reply:
But we can't give it the net driver in tests. It shouldn't really have a net driver at all except the existing code means there is little choice.

Reply:
Cool, I think the counter argument is that having this list on all workers in unintuitive and might end up confusing someone during debugging, but facilitating tests sounds like a winning reason.

Reply:
Unfortunately, there are a lot of things in the VirtualWorkerTranslator which are confusing because sometimes it's in Authoritative mode and sometimes not. I'm half tempted to split the entire thing, but I don't think that's a path we should go down now.